keanu-project/src/main/java/io/improbable/keanu/tensor/validate/policy/ChangeValueTo.java

```diff
@@ -0,0 +1,31 @@
+package io.improbable.keanu.tensor.validate.policy;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+
+public class ChangeValueTo<DATATYPE, TENSOR extends Tensor<DATATYPE>> implements TensorValidationPolicy<TENSOR> {
+    private final DATATYPE value;
+
+    // package private - because it's created by the factory method TensorValidationPolicy.changeValueTo
+    ChangeValueTo(DATATYPE value) {
+        this.value = value;
+    }
+
+    @Override
+    public TENSOR handle(TENSOR tensor, BooleanTensor result) {
+        if (tensor instanceof DoubleTensor) {
```

Comment:
There is a `setWithMask` on `NumberTensor`, but not on `Tensor`. I might be able to make that work, but it entails other changes: the `handle` method would need to take a `TENSOR mask` instead of `BooleanTensor result`. This changes the interface so has an impact on the other classes related to `TensorValidator`.